### PR TITLE
Fixed hash serializer for inline serialization

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Replace deprecated `addData` function with `setData` function to avoid a serialization error when serialize [inline](https://jmsyst.com/libs/serializer/master/reference/annotations#inline). ~~Replace deprecated [GenericSerializationVisitor](https://github.com/schmittjoh/serializer/blob/1.13.0/src/JMS/Serializer/GenericSerializationVisitor.php#L11)~~

#### Why?

To fix hash serializer for inline serialization. If you have subentities which are also auditable and you try to inline them it will error with `_hash` exist instead. 

#### Example Usage

~~~php
    /**
     * @Serializer\Expose
     * @Serializer\Inline
     *
     * @ORM\OneToOne(targetEntity="App\Model\Location\LocationDimensionRestaurant", mappedBy="locationDimension", cascade={"persist", "remove"})
     */
    private $restaurant;

    /**
     * @Serializer\Expose
     * @Serializer\Inline
     *
     * @ORM\OneToOne(targetEntity="App\Model\Location\LocationDimensionBaker", mappedBy="locationDimension", cascade={"persist", "remove"})
     */
    private $baker;
~~~

### CODE REVIEW

File: src/Sulu/Component/Hash/Serializer/Subscriber/HashSerializeEventSubscriber.php [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 use JMS\Serializer\EventDispatcher\EventSubscriberInterface;
 use JMS\Serializer\EventDispatcher\ObjectEvent;
-use JMS\Serializer\GenericSerializationVisitor;
+use JMS\Serializer\JsonSerializationVisitor;
```
Comment (danrot) [inline src/Sulu/Component/Hash/Serializer/Subscriber/HashSerializeEventSubscriber.php]: Is that change required? This means this subscriber is only working for JSON serialization...

File: src/Sulu/Component/Serializer/ArraySerializationVisitor.php [hunk 2]
```diff
@@ -11,12 +11,12 @@
 
 namespace Sulu\Component\Serializer;
 
-use JMS\Serializer\GenericSerializationVisitor;
+use JMS\Serializer\JsonSerializationVisitor;
```
Comment (danrot) [inline src/Sulu/Component/Serializer/ArraySerializationVisitor.php]: Same as above.

File: src/Sulu/Component/Hash/Serializer/Subscriber/HashSerializeEventSubscriber.php (hunk 1 above)
Comment (alexander-schranz) [reply src/Sulu/Component/Hash/Serializer/Subscriber/HashSerializeEventSubscriber.php]: currently not required but required when we will upgrade to 2.0

File: src/Sulu/Component/Serializer/ArraySerializationVisitor.php (hunk 2 above)
Comment (alexander-schranz) [reply src/Sulu/Component/Serializer/ArraySerializationVisitor.php]: did undo the change and still use the GenericSerializationVisitor until we upgrade
